server/src/main/java/org/opensearch/indices/IndicesRequestCache.java

```diff
@@ -276,7 +301,7 @@ public boolean equals(Object o) {
             if (this == o) return true;
             if (o == null || getClass() != o.getClass()) return false;
             Key key = (Key) o;
-            if (Objects.equals(readerCacheKey, key.readerCacheKey) == false) return false;
+            if (Objects.equals(readerCacheKeyUniqueId, key.readerCacheKeyUniqueId) == false) return false;
```

Comment:
nit: Probably unique is implicit in `readerCacheKeyUniqueId`. So, can be better written as `readerCacheKeyId`

Reply:
Make sense.